### ISSUE TITLE
Compile error for Android

### ISSUE THREAD

**Issue by ytain** [mosra/magnum#131]
Hello mosra,

I'm getting a compile error for Android target. Tried various ways to try to fix it, with no luck.

The error is below:
```
Scanning dependencies of target Magnum
[  2%] Building CXX object src/Magnum/CMakeFiles/Magnum.dir/AbstractFramebuffer.cpp.o
<command-line>:0:9: error: expected identifier before numeric constant
/Volumes/xxxx/xxxx/temporary/magnum/src/Magnum/Extensions.h:1:15: note: in expansion of macro 'ANDROID'
 #ifndef Magnum_Extensions_h
               ^
<command-line>:0:9: error: expected unqualified-id before numeric constant
/Volumes/xxxx/xxxx/temporary/magnum/src/Magnum/Extensions.h:1:15: note: in expansion of macro 'ANDROID'
 #ifndef Magnum_Extensions_h
               ^
/Volumes/xxxx/xxxx/temporary/magnum/src/Magnum/AbstractFramebuffer.cpp:495:1: error: expected '}' at end of input
 }
 ^
/Volumes/xxxx/xxxx/temporary/magnum/src/Magnum/AbstractFramebuffer.cpp:495:1: error: expected '}' at end of input
/Volumes/xxxx/xxxx/temporary/magnum/src/Magnum/AbstractFramebuffer.cpp:495:1: error: expected '}' at end of input
make[2]: *** [src/Magnum/CMakeFiles/Magnum.dir/AbstractFramebuffer.cpp.o] Error 1
make[1]: *** [src/Magnum/CMakeFiles/Magnum.dir/all] Error 2
make: *** [all] Error 2
```

**Comment by mosra** [comment]
At first I was like "what the fuck, this is not possible", but I guess I might know where is the problem! :)

Can you put the following at the top of the Extensions.h header?
```cpp
#undef ANDROID
```

Not sure where the definition comes from. Isn't it possible that you are somehow defining such preprocessor macro? I just checked locally and it works flawlessly. I have quite an old NDK version though (r9c). What's your SDK/NDK version?

**Comment by mosra** [comment]
Update: aargh, not again. It seems that `ANDROID` is defined by the *standard* build system: http://stackoverflow.com/a/6375823 Which poor soul thought that defining such unprefixed macro is a good idea? Dammit. Lately it seems that I'm only fixing preprocessor madness done by incompetent people, see mosra/corrade#19. Unbelievable.

However, I tried to *not* use the standard build system to avoid these kinds of fuckups so the macro *should not* get this far. Can you describe your setup in more detail?

**Comment by ytain** [comment]
By using my own shell script to automate building the magnum libraries and dependencies found there at https://gist.github.com/536a7d1ab9fd28ce6362

Prior to this you also need to download the polly repo from https://github.com/ruslo/polly cause it contains most of the proper custom toolchain.cmake files for various custom targets.

You can adjust accordingly the shell script to accomodate the folder structures. The variable temporary_folder is the root of all the folders that contains the magnum repos and the dependencies like libpng, freetype2, harfbuzz, zlib etc.

I'm using the Android NDK 10e which is also located in the same temporary_folder location as you can see from the shell script.

You are welcome to use the shell script to make your own CI scripts for automatic build of libraries for custom targets like Android/iOS.

**Comment by mosra** [comment]
OK, right, I see.

Would it be a big problem if I just `#undef`'d the `ANDROID` macro in the Extensions.h header? I'm using the ANDROID identifier for extension names, like `Extensions::GL::ANDROID::extension_pack_es31a` and I really don't want to make it different just because someone at Google decided that `#define ANDROID` is a good practice ;)

**Comment by ytain** [comment]
I don't know if you can use ANDROID_ENV to signify that it is in an Android environment instead of the regular ANDROID?

**Comment by mosra** [comment]
Corrade provides [CORRADE_TARGET_ANDROID](http://mosra.cz/blog/corrade-doc/Corrade_8h.html#a4d1ec6acb651e40f04ad8aef103f51a2) and similar consistently named defines for other platforms, would that be sufficient for you?

**Comment by ytain** [comment]
Yes. That would be enough.

**Comment by mosra** [comment]
Fixed in 281bcac62413e17067041d35944becd55d383ffd. Thanks for reporting this!
